test/pool.test.ts

```diff
@@ -28,6 +28,41 @@ async function mineBlocksUntilCycleEnd(): Promise<void> {
 
 type AnyPool = EthPool | Erc20Pool;
 
+class ProxyReceiverWeight {
+  receiverWeight: number;
+  proxyWeight: number;
+
+  constructor(receiverWeight: number, proxyWeight: number) {
+    this.receiverWeight = receiverWeight;
+    this.proxyWeight = proxyWeight;
+  }
+
+  static empty(): ProxyReceiverWeight {
+    return new ProxyReceiverWeight(0, 0);
+  }
+
+  isEmpty(): boolean {
+    return this.receiverWeight == 0 && this.proxyWeight == 0;
+  }
+}
+
+type ProxyReceiverWeights = Map<string, ProxyReceiverWeight>;
+
+function updateReceiversList(
+  list: ProxyReceiverWeights,
+  receiver: PoolUser,
```

Comment:
We don’t need any information from `PoolUser` except for the address.
```suggestion
  receiver: string,
```
I’d prefer this approach because the function signature looks much more like your usual update function with object-key-function arguments.

Reply:
Fixed